test/cli/install/bun-upgrade.test.ts

```diff
@@ -0,0 +1,116 @@
+import { spawn, spawnSync } from "bun";
+import { afterEach, beforeEach, expect, it, describe } from "bun:test";
+import { bunExe, bunEnv as env } from "harness";
+import { mkdtemp, realpath, rm, mkdir, stat } from "fs/promises";
+import { tmpdir } from "os";
+import { join } from "path";
+import { cpSync } from "js/node/fs/export-star-from";
+
+let run_dir: string;
+
+beforeEach(async () => {
+  run_dir = await realpath(
+    await mkdtemp(join(tmpdir(), "bun-upgrade.test." + Math.trunc(Math.random() * 9999999).toString(32))),
+  );
+  cpSync(bunExe(), run_dir);
+});
+afterEach(async () => {
+  await rm(run_dir, { force: true, recursive: true });
+});
+
+it("two invalid arguments, should display error message and suggest command", async () => {
+  const { stderr, exitCode } = spawn({
+    cmd: [bunExe(), "upgrade", "bun-types", "--dev"],
```

Comment:
bunExe() is the absolute path to the bin binary so this doesnt work, you need to pass the path to the copied executable

Reply:
fixed via commit https://github.com/oven-sh/bun/pull/7784/commits/e0e76c436aaf5825947ae85a5661b1f097342d51